src/herder/test/HerderTests.cpp

```diff
@@ -1620,100 +1620,126 @@ TEST_CASE("SCP State", "[herder][acceptance]")
+    auto A = simulation->getNode(validatorAKey.getPublicKey());
+    auto B = simulation->getNode(validatorBKey.getPublicKey());
+    auto C = simulation->getNode(validatorCKey.getPublicKey());
 
-    auto currentValidatorLedger = [&]() {
-        return validator->getLedgerManager().getLastClosedLedgerNum();
+    auto currentALedger = [&]() {
+        return A->getLedgerManager().getLastClosedLedgerNum();
     };
-    auto currentListenerLedger = [&]() {
-        return listener->getLedgerManager().getLastClosedLedgerNum();
+    auto currentCLedger = [&]() {
+        return C->getLedgerManager().getLastClosedLedgerNum();
     };
 
     auto waitForLedgers = [&](int nLedgers) {
-        auto destinationLedger = currentValidatorLedger() + nLedgers;
+        auto destinationLedger = currentALedger() + nLedgers;
         simulation->crankUntil(
             [&]() {
                 return simulation->haveAllExternalized(destinationLedger, 100);
             },
             2 * nLedgers * Herder::EXP_LEDGER_TIMESPAN_SECONDS, false);
-        return std::min(currentListenerLedger(), currentValidatorLedger());
+        return std::min(currentALedger(), currentCLedger());
     };
 
-    auto waitForValidator = [&](int nLedgers) {
-        auto destinationLedger = currentValidatorLedger() + nLedgers;
+    auto waitForA = [&](int nLedgers) {
+        auto destinationLedger = currentALedger() + nLedgers;
         simulation->crankUntil(
-            [&]() { return currentValidatorLedger() >= destinationLedger; },
+            [&]() { return currentALedger() >= destinationLedger; },
             2 * nLedgers * Herder::EXP_LEDGER_TIMESPAN_SECONDS, false);
-        return currentValidatorLedger();
+        return currentALedger();
     };
 
     uint32_t currentLedger = 1;
-    REQUIRE(currentValidatorLedger() == currentLedger);
-    REQUIRE(currentListenerLedger() == currentLedger);
+    REQUIRE(currentALedger() == currentLedger);
+    REQUIRE(currentCLedger() == currentLedger);
 
-    // externalize a few ledgers
-    auto maxSlots = validator->getConfig().MAX_SLOTS_TO_REMEMBER;
-    currentLedger = waitForLedgers(maxSlots / 2);
+    // All nodes externalize a few ledgers
+    auto fewLedgers = A->getConfig().MAX_SLOTS_TO_REMEMBER / 2;
+    currentLedger = waitForLedgers(fewLedgers);
 
-    REQUIRE(currentValidatorLedger() >= currentLedger);
-    // listener is at most a ledger behind
-    REQUIRE(currentLedger == currentListenerLedger());
+    // C is at most a ledger behind
+    REQUIRE(currentALedger() >= currentLedger);
+    REQUIRE(currentCLedger() == currentLedger);
 
-    // disconnect listener
-    simulation->dropConnection(validatorKey.getPublicKey(),
-                               listenerKey.getPublicKey());
+    // disconnect C
+    simulation->dropConnection(validatorAKey.getPublicKey(),
+                               validatorCKey.getPublicKey());
+
+    HerderImpl& AHerder = *static_cast<HerderImpl*>(&A->getHerder());
```

Comment:
names should be things like `herderA` (this also avoids not following our coding standard for variable names)